docs/content/docs/dev/table/hiveCompatibility/hiveDialect/overview.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "Overview"
+weight: 1
+type: docs
+aliases:
+- /dev/table/hiveCompatibility/hiveDialect/overview
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Hive Dialect
+
+Flink allows users to write SQL statements in Hive syntax when Hive dialect is used.
+By providing compatibility with Hive syntax, we aim to improve the interoperability with Hive and reduce the scenarios when users need to switch between Flink and Hive in order to execute different statements.
+
+## Use Hive Dialect
+
+Flink currently supports two SQL dialects: `default` and `hive`. You need to switch to Hive dialect
+before you can write in Hive syntax. The following describes how to set dialect with
+SQL Client and Table API. Also notice that you can dynamically switch dialect for each
+statement you execute. There's no need to restart a session to use a different dialect.
+
+{{< hint warning >}}
+**Note:**
+
+- To use Hive dialect, you have to add dependencies related to Hive. Please refer to [Hive dependencies]({{< ref "docs/connectors/table/hive/overview" >}}#dependencies) for how to add the dependencies.
```

Comment:
Do we need to mention it in this "Hive dependencies" part? Only when using Hive dialect, do we need to switch the planner jar.
I think the "Hive dependencies" part is just to tell user what extra jars needed when they want to use Hive as a source or sink.  If we mention it in "Hive dependencies" part, it may mislead users always to switch the planner jar. 
Or we add another small  part like "When you want to use dialect, you need to switch the planner jar " in "Hive dependencies" part?

Reply:
I think we should mention this in "Hive dependencies" because this is also shared for using Hive dialect. You can add a note that the planner jar switching is only needed when using Hive dialect and HiveServer2 Endpoint, but this is the recommended setup for Hive integration.

Reply:
Could you open another PR and issue for this improvement?

Reply:
sure, I have create [FLINK-29350](https://issues.apache.org/jira/browse/FLINK-29350) for it and will open another pr assp.